source/features/filter-pr-by-build-status.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import {checkInline} from '../libs/icons';
+import features from '../libs/features';
+
+function init() {
+	const statusFilter = select('.table-list-header-toggle > details:nth-last-child(3)')!.cloneNode(true) as HTMLDetailsElement;
+
+	select('summary', statusFilter)!.textContent = 'Status\u00A0';
+	select('.select-menu-title', statusFilter)!.textContent = 'Filter by build status';
+	select('.select-menu-list', statusFilter)!.innerHTML = '';
+
+	select('.table-list-header-toggle > details:nth-last-child(2)')!.before(statusFilter);
+	select('.table-list-header-toggle > details:nth-last-child(3)')!.addEventListener('toggle', populateDropDown, {once: true});
+}
+
+function populateDropDown() {
+	const statusList = ['Success', 'Failure', 'Pending'];
+
+	const searchParam = new URLSearchParams(location.search);
+	let queryString = searchParam.get('q') || '';
+
+	const [, currentStatus] = queryString.match(/status:(success|failure|pending)/) || ['', ''];
```

Comment:
```suggestion
	const [, currentStatus = ''] = queryString.match(/status:(success|failure|pending)/);
```

Is this possible? Would be nicer.

Reply:
You should use `\b` to match on word boundaries to prevent false positives.

Reply:
It's definitely better. Thanks!

Reply:
Oops, sorry for pushing and pulling on this line, but the ` || []` at the end is required. 

Try merging with master to ensure that whatever we try is strict TypeScript